core/src/main/scala/org/ensime/indexer/FileWatchers.scala

```diff
@@ -79,10 +144,10 @@ class ClassfileWatcher(
         running = scheduler.schedule(resetter, 1, TimeUnit.SECONDS)
       }
 
-      private val targets: Set[FileName] = for {
-        dir <- config.targetClasspath
+      private val targets: Set[FileName] = (for {
+        dir <- watchedDirs
         ref = vfs.vfile(dir)
-      } yield ref.getName()
+      } yield ref.getName()).toSet
```

Comment:
it should be slightly more efficient to take a Set[File] as watchedDirs (avoids an intermediate collection)